concepts/0003-protocols/README.md

```diff
@@ -185,29 +217,263 @@ A protocol has the following ingredients:
-also attached to this RFC as an example.
+protocol concepts, are given in the [instructions for protocol RFCs](#instructions-for-protocol-rfcs)
+and in the [protocol RFC template](../../0000-template-protocol.md).
+
+The [tictactoe protocol](tictactoe/README.md) is attached to this RFC as an example.
 
 [![tictactoe protocol](tictactoe/tile.png)](tictactoe/README.md)
 
-#### Security Considerations
+### Security Considerations
 
-##### Replay attacks
+#### Replay attacks
 
-It should be noted that when defining a protocol that has domain specific requirements around preventing replay attacks an `@id` property SHOULD be required. Given the `@id` field is most commonly set to be a UUID, it usually provides sufficient randomness that a nonce would in preventing replay attacks. This means that sufficient care will be needed in processing of the `@id` field however, to make sure the `@id` field hasn't been used before. In some cases, nonces require being unpredictable as well. In this case, greater review should be taken as to how the `@id` field should be used in the domain specific protocol. Additionally, in the event where the `@id` field is not adequate, it's recommended that an additional `nonce` field be required by the domain specific protocol specification.
+It should be noted that when defining a protocol that has domain specific requirements around preventing replay attacks, an `@id` property SHOULD be required. Given an `@id` field is most commonly set to be a UUID, it should provide randomness comparable to that of a nonce in preventing replay attacks. However, this means that care will be needed in processing of the `@id` field to make sure its value has not been used before. In some cases, nonces require being unpredictable as well. In this case, greater review should be taken as to how the `@id` field should be used in the domain specific protocol. In the event where the `@id` field is not adequate for preventing replay attacks, it's recommended that an additional `nonce` field be required by the domain specific protocol specification.
 
 ## Reference
 
-* [Message Type and Protocol Identifier URIs](uris.md)
-* [Semver Rules for Protocols](semver.md)
-* [State Details and State Machines](state-details.md)
-* [Roles, Participants, Parties, and Controllers](roles-participants-etc.md)
+### Message Type and Protocol Identifier URIs
+
+Message types and protocols are identified with URIs that match certain
+conventions.
+
+#### MTURI
+
+A __message type URI__ (MTURI) identifies message types unambiguously.
+Standardizing its format is important because it is parsed by agents that
+will map messages to handlers--basically, code will look at this string and
+say, "Do I have something that can handle this message type inside protocol
+*X* version *Y*?" When that analysis happens, it must do more than compare
+the string for exact equality; it may need to check for semver compatibility.
```

Comment:
I think this phrase "more than compare the string for exact equality" accurately reflects the intent of the community after our last call, but we need to be more explicit about the comparison algorithm. We also have to be careful because semver compatibility and exact equality are contradictory concepts; semver is about predictable fuzziness.

URIs are case-sensitive after a domain name, but case-insensitive before. Per internet specs, "https://foo.com/myprotocol/1.0" and "https://FOO.COM/myprotocol/1.0" are identical URIs. "Exact equality" means something different in unicode than it does in normal human intuition (there are several byte sequences in UTF8 that map to lower-case-latin-e-with-acute-accent; all will render as visually identical, and all are considered normalized).

I suggest something like this:

Keep "A __message type URI___ ...*X* version *Y*?" as written. Then add a paragraph break, followed by:

"When this analysis happens, strings should be compared for byte-wise equality in all segments except version. This means that case, unicode normalization, and punctuation differences all matter. It is thus best practice to avoid protocol and message names that differ only in subtle, easy-to-mistake ways."

"Comparison of the version segment of an MTURI or PIURI should follow semver rules and is discussed elsewhere."

Reply:
Nice.  Changed.